superset/hive_query.py

```diff
@@ -1,36 +1,13 @@
-"""The file  Generate Hive DB partition based SQL query for any selected time range  and defined gran from superset UI,
-   
-    Use  below  data structure for extra_params in db metadata Extra section to enable partitions for hive
-     "extra_params":{
-        "partitions":{
-         "year":"year",
-         "month":"month",
-         "day":"day",
-         "hour":"hour",
-         "minute":"minute"
-         }
-    }
-
-"""
 import json
 import re
 from datetime import timedelta
-
-#gran valaue map
-GRAN_VALUE_MAP =  {
+import logging
+from datetime import datetime
+#gran valaue map,supported grain by hive db engine
+GRAN_VALUE_MAP = {
     'PT1S' : 1,
-    'PT5S' : 5,
-    'PT10S' :10,
-    'PT15S' :15,
-    'PT30S': 30,
     'PT1M' : 60,
-    'PT5M' : 300,
-    'PT10M': 600,
-    'PT15M': 900,
-    'PT30M': 1800,
-    'PT0.5H':1800,
     'PT1H' : 3600,
-    'PT6H' : 21600,
     'P1D'  : 86400,
     'P1W'  : 604800,
     'P1M'  : 2629743,
```

Comment:
What is value of 1M. usaly we take 30 days that works out to be 2592000 seconds

Reply:
currently i m using conversion as follows as per https://www.epochconverter.com/

Human readable time | Seconds
-- | --
1 hour | 3600 seconds
1 day | 86400 seconds
1 week | 604800 seconds
1 month (30.44 days) | 2629743 seconds
1 year (365.24 days) | 31556926 seconds


we can change as per our use case if required.

Reply:
If we can then we should use out standard 30day for month and 365 day for year.
I thought it is govern by engine.

Reply:
currently i m updating as 30day for month and 365 day for year , later we will see  how it relates with engine